Gems/LyShine/Code/Editor/ViewportWidget.cpp

```diff
@@ -886,17 +910,47 @@ void ViewportWidget::OnFontTextureUpdated([[maybe_unused]] IFFont* font)
     m_fontTextureHasChanged = true;
 }
 
+LyShine::AttachmentImagesAndDependencies ViewportWidget::GetRenderTargets()
+{
+    LyShine::AttachmentImagesAndDependencies canvasTargets;
+
+    AZ::EntityId canvasEntityId;
+    UiEditorMode editorMode = m_editorWindow->GetEditorMode();
+    if (editorMode == UiEditorMode::Edit)
+    {
+        canvasEntityId = m_editorWindow->GetCanvas();
+    }
+    else // if (editorMode == UiEditorMode::Preview)
+    {
+        canvasEntityId = m_editorWindow->GetPreviewModeCanvas();
+    }
```

Comment:
If seems this block of code is only relevant to m_editorWindow. Does it make sense to wrap it to a function for m_editorWindow. For example: m_editorWindow->GetActiveCanvas()